File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = prices[0], profit = 0;

        for(int num: prices){
            if(num < buy){
                buy = num;
            }

            if(num - buy > profit){
                profit = num - buy;
            }
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int b = prices[0], s = 0;

        for(int i=1; i<prices.size(); i++){
            if(b>prices[i]){
                b = prices[i];
            }
            
            if(prices[i] - b > s){
                s = prices[i] - b;
            }
        }
        return s;
    }
};